x-pack/plugins/security_solution/server/search_strategy/index_fields/index.test.ts

```diff
@@ -780,3 +790,119 @@ describe('Index Fields', () => {
     });
   });
 });
+
+describe('Fields Provider', () => {
+  describe('search', () => {
+    const getFieldsForWildcardMock = jest.fn();
+    const esClientSearchMock = jest.fn();
+
+    const deps = ({
+      esClient: { asCurrentUser: { search: esClientSearchMock } },
+    } as unknown) as SearchStrategyDependencies;
+
+    beforeAll(() => {
+      getFieldsForWildcardMock.mockResolvedValue([]);
+      IndexPatternsFetcher.prototype.getFieldsForWildcard = getFieldsForWildcardMock;
+    });
+
+    beforeEach(() => {
+      getFieldsForWildcardMock.mockClear();
+      esClientSearchMock.mockClear();
+    });
+
+    afterAll(() => {
+      getFieldsForWildcardMock.mockRestore();
+    });
+
+    it('should check index exists', async () => {
+      const indices = ['some-index-pattern-*'];
+      const request = {
+        indices,
+        onlyCheckIfIndicesExist: true,
+      };
+
+      const response = await requestIndexFieldSearch(request, deps, beatFields);
+
+      expect(getFieldsForWildcardMock).toHaveBeenCalledWith({ pattern: indices[0] });
+
+      expect(response.indexFields).toHaveLength(0);
+      expect(response.indicesExist).toEqual(indices);
+    });
+
+    it('should search index fields', async () => {
+      const indices = ['some-index-pattern-*'];
+      const request = {
+        indices,
+        onlyCheckIfIndicesExist: false,
+      };
+
+      const response = await requestIndexFieldSearch(request, deps, beatFields);
+
+      expect(getFieldsForWildcardMock).toHaveBeenCalledWith({ pattern: indices[0] });
+
+      expect(response.indexFields).not.toHaveLength(0);
```

Comment:
can we explicitly say how many indices this test should return rather than `not 0`?

Reply:
Here I just wanted to test that if the request has the parameter `onlyCheckIfIndicesExist: true`, the `indexFields` is empty. Otherwise, it returns the fields.
But I didn't want to check the `indexFields` content explicitly because it is the output of the `formatIndexFields` function, which adds a lot of static _beatFields_ and others, and this function is already tested above, so I didn't want to re-test that again.

Maybe it would be better if I do: `expect(response.indexFields.length).toBeGreaterThan(0);`? 
Or if you think I should check the exact number I can try to do some calculations.

Reply:
Ahh, the `formatIndexFields` test above was hidden in the diff. 👍🏾 The way you have it is fine as it can't have a negative length 😄